### DISCUSSION TITLE
A little help (with colors)

### DISCUSSION THREAD

**Discussion by Ctrl-Alb** [fastfetch-cli/fastfetch#1379] [Q&A] [answered]
Hi!
I'm trying to customize my Fastfetch config file and I found the wiki and the "examples" folder very useful.
However, I'm still struggling with three specific things and hope someone can help me:

1. I would like to display certain percentages (like brightness) without the green-yellow-red color scheme, using just the regular text color instead.
![Screenshot 1](https://github.com/user-attachments/assets/83854df6-792c-4adf-adcf-0897a5a052ca)

4. I would like the separator to match the "keyColor" and the key width (bold). Currently, the only way I can achieve this is by leaving the "separator" blank (" ") and manually adding "->" in front of each key title. While this gives me the aesthetic I want (see the attached image), I'm wondering if there's a more straightforward solution.
![Screenshot 2](https://github.com/user-attachments/assets/2293ef7e-f2e2-4362-b69f-e39fce308655)

5. I would like the line under the title to match both the color (green) and the width (bold) of the title.
![Screenshot 2](https://github.com/user-attachments/assets/22ecd3a3-9d3f-4022-b36e-a63f21c30a0c)

Here's my config file:

```
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",

    "logo": {
       "padding": {
          "top": 5
        }
    },

    "display": {
       "separator": " -> ",
       "color": {
          "separator": "1"
        }
    },

    "modules": [

       {
           "type": "title",
           "color": {
              "user": "green",
              "at": "green",
              "host": "green"
           }
       },

       "separator",

        "break",
       {
          "type": "custom",
          "format": "{#keys}{#1}{#yellow}모Hardware"
       },
       {
          "type": "host",
          "key": "├ Model ",
          "keyColor": "yellow",
          "format": "{5} {2} {3}"
       },
       {
          "type": "display",
          "key": "├ Screen",
          "keyColor": "yellow",
          "format": "{1} x {2} px @ {3} Hz in {12}\" ({13} ppi) - {14} bit - HDR: {20} - {7}"
       },
       {
          "type": "brightness",
          "key": "├ Bright",
          "keyColor": "yellow"
       },
       {
          "type": "cpu",
          "key": "├ CPU   ",
          "keyColor": "yellow",
          "format": "{1} - {5} core @ {7} - {8}",
          "temp": true
       },
       {
          "type": "gpu",
          "key": "├ GPU   ",
          "keyColor": "yellow",
          "format": "{2} - {5} core @ {12} - {4} - {11}",
          "temp": true
       },
       {
          "type": "memory",
          "key": "├ Memory",
          "keyColor": "yellow"
       },
       {
          "type": "disk",
          "key": "├ Disk  ",
          "keyColor": "yellow",
          "format": "{10} - {1} / {2} ({3}) - {9}"
        },
        {
          "type": "battery",
          "key": "├ Charge",
          "keyColor": "yellow",
          "format": "{4} ({12} hours, {13} mins remaining) - {7} cycles - {6}",
          "temp": true,
          "percent": {
             "green": 51,
             "yellow": 11
          }
       },
       {
          "type": "camera",
          "key": "├ Camera",
          "keyColor": "yellow",
          "format": "{1} - {5} x {6} px"
       },
       {
          "type": "sound",
          "key": "└ Sound ",
          "keyColor": "yellow",
          "format": "{2} - {3}"
       },
        "break",
       {
          "type": "custom",
          "format": "{#keys}{#1}{#magenta} Software"
       },
       {
          "type": "os",
          "key": "├ OS Ver",
          "keyColor": "magenta",
          "format": "{3} {10} {8} - {12}"
       },
       {
          "type": "uptime",
          "key": "├ Uptime",
          "keyColor": "magenta"
       },
       {
          "type": "shell",
          "key": "├ Shell ",
          "keyColor": "magenta"
       },
       {
          "type": "processes",
          "key": "└ Tasks ",
          "keyColor": "magenta"
       },
        "break",
       {
          "type": "custom",
          "format": "{#keys}{#1}{#blue}ᯤ Connectivity"
       },
       {
          "type": "publicip",
          "key": "├ Pub IP",
          "keyColor": "blue",
          "format": "{1} - {2}"
       },
       {
          "type": "localip",
          "key": "├ Loc IP",
          "keyColor": "blue",
          "format": "{1} - {3}",
          "showMac": true
       },
       {
          "type": "dns",
          "key": "├ DNS   ",
          "keyColor": "blue"
       },
       {
          "type": "wifi",
          "key": "├ Wi-Fi ",
          "keyColor": "blue",
          "format": "{4} - {7} - {6} - {10}"
       },
       {
          "type": "bluetooth",
          "key": "├ BT Dev",
          "keyColor": "blue",
          "format": "{1} - {4}"
       },
       {
          "type": "bluetoothradio",
          "key": "└ BT Ver",
          "keyColor": "blue",
          "format": "{5}"
       },
        "break"
    ]
}
```

Thank you for your help

**Comment by CarterLi** [comment]
1. There is `--percent-type`. See `fastfetch -h percent-type`
2. `1;3x` should do the trick. See `fastfetch -h color`
3. There is `--separator-output-color`

`fastfetch -h` provides a lot of useful information. You should read it first

**Reply by Ctrl-Alb** [reply]
Hi @CarterLi 
Thank you for your reply. 
Unfortunately my JavaScript skills are very limited, which is why I'm having trouble with my configuration.
1. Regarding `--percent-type` , I have already searched for that specific command and tried several attempts, but without success.
As I mentioned, I don't want to display all percentages in the regular text color, just some of them.
I tried to add `--percent-type` to individual categories, but it didn't work (see the examples).
Ex.
```
       {
          "type": "sound",
          "key": "└ Sound ",
          "keyColor": "yellow",
          "format": "{2} - {3}",
          "percent-type": 1
       },
```
Ex.
```
       {
          "type": "sound",
          "key": "└ Sound ",
          "keyColor": "yellow",
          "format": "{2} - {3}",
          "percent": {
             "percent-type": 1
          }
       },
```
2. I'm using three different `keyColor` values (as you can see in the second screenshot: yellow, magenta and blue).
As far as I understand, using `1;3x` would color all separators with the same color. 
However, my question is whether it's possible for the separators to automatically match the `keyColor` (as shown in the second screenshot).
3. I don't know how to use `--separator-output-color`  , my attempts have been unsuccessful (see the example)
Ex.
```
       "separator": {
              "separator-output-color "1;green"
           }
```

**Comment by CarterLi** [comment]
I was on a vacation.

1.
```json
{
    "display": {
        "percent": {
            "type": 1
        }
    }
}
```

2. > whether it's possible for the separators to automatically match the keyColor (as shown in the second screenshot).

No

3.
```json
{
    "type": "separator",
    "outputColor": "bright_green"
}
```

**Reply by Ctrl-Alb** [reply]
Hi @CarterLi ,
I hope you had a good vacation :)
1. If I understand correctly, this would display all percentages in the default text format.
However, as I mentioned earlier, I was looking for a way to display only specific percentages (e.g., brightness and sound) in the default text format, rather than all percentages.
2. Ok, no worries
3. Thanks, it worked!

**Reply by CarterLi** [reply]
It's not supported to set percent format in module level.

**Reply by Ctrl-Alb** [reply]
Thanks for the clarification.
Since it's already possible:
- to customize percentages per-module (for example, defining thresholds for red, yellow, and green),
- to display (all) percentages in different formats,

Do you think it would be possible in the future to allow customization of the percentage format at the module level?

**Reply by CarterLi** [reply, marked as answer]
1. Threshold configuration was already supported.
2. Added in dev branch.

Please test! JSON Config example:

```jsonc
{
    "type": "memory",
    "percent": {
        "green": 20, // [0%, 20%) will be displayed in green
        "yellow": 40, // [20, 40) will be displayed in yellow and [40, 100] will be displayed in red
        "type": [ // Display percent value in monochrome bar, same as 10
            "bar",
            "bar-monochrome"
        ]
    }
}
```

**Reply by Ctrl-Alb** [reply]
Hey @CarterLi 
1. Yes, exactly. I was trying to say that since thresholds were already supported per-module, maybe the format per-module could be added, too. And you just added it, great!
2. Awesome! I tried it and it worked perfectly

Thank you again for your continuous efforts in improving Fastfetch :)
